File: src/Havenly.Domain/Aggregates/Owner.cs
using Domain.Common.Models;
using Domain.Entities;
using Domain.ValueObjects;

namespace Domain.Aggregates;

public class Owner : AggregateRoot<OwnerId>
{
    public UserId UserId { get; set; }
    public User User { get; set; } = null!;

    public List<Property> Properties { get; set; } = [];

    private Owner(OwnerId id, UserId userId) : base(id)
    {
        UserId = userId;
    }

    public Owner()
    {
    }

    public static Owner Create(UserId userId)
    {
        return new Owner(OwnerId.CreateUnique(), userId);
    }
}

File: src/Havenly.Web/src/app/common/owner.ts
// public UserId UserId { get; set; }
// public User User { get; set; } = null!;
//
// public List<Property> Properties { get; set; } = [];

import {Property} from "./property";

export class Owner {
  constructor(public Id: string,
              public UserId: string,
              public Name: string,
              public Email: string,
              public ContactInfo: string,
              public Properties: Property[]) {
  }
}
